src/goto-instrument/contracts/dynamic-frames/dfcc_contract_clauses_codegen.cpp

```diff
@@ -0,0 +1,288 @@
+/*******************************************************************\
+
+Module: Dynamic frame condition checking for function contracts
+
+Author: Remi Delmas, delmasrd@amazon.com
+Date: February 2023
+
+\*******************************************************************/
+#include "dfcc_contract_clauses_codegen.h"
+
+#include <util/expr_util.h>
+#include <util/fresh_symbol.h>
+#include <util/invariant.h>
+#include <util/mathematical_expr.h>
+#include <util/namespace.h>
+#include <util/pointer_offset_size.h>
+#include <util/std_expr.h>
+
+#include <goto-programs/goto_model.h>
+
+#include <ansi-c/c_expr.h>
+#include <goto-instrument/contracts/utils.h>
+#include <langapi/language_util.h>
+
+#include "dfcc_library.h"
+#include "dfcc_spec_functions.h"
+#include "dfcc_utils.h"
+
+dfcc_contract_clauses_codegent::dfcc_contract_clauses_codegent(
+  goto_modelt &goto_model,
+  message_handlert &message_handler,
+  dfcc_utilst &utils,
+  dfcc_libraryt &library,
+  dfcc_spec_functionst &spec_functions)
+  : goto_model(goto_model),
+    message_handler(message_handler),
+    log(message_handler),
+    utils(utils),
+    library(library),
+    spec_functions(spec_functions),
+    ns(goto_model.symbol_table)
+{
+}
+
+void dfcc_contract_clauses_codegent::gen_spec_assigns_instructions(
+  const irep_idt &language_mode,
+  const exprt::operandst &assigns_clause,
+  goto_programt &dest)
+{
+  for(const auto &expr : assigns_clause)
+  {
+    if(can_cast_expr<conditional_target_group_exprt>(expr))
+    {
+      encode_assignable_target_group(
+        language_mode, to_conditional_target_group_expr(expr), dest);
```

Comment:
You might want to use `if(auto target_group = expr_try_dynamic_cast<conditional_target_group_exprt>(expr))` and then just use `*target_group` instead of `to_conditional_target_group_expr(expr)`.

Reply:
fixed